modules/aioseop_sitemap.php

```diff
@@ -443,13 +443,40 @@ public function __construct() {
 			add_filter( $this->prefix . 'display_options', array( $this, 'filter_display_options' ) );
 			add_filter( $this->prefix . 'update_options', array( $this, 'filter_options' ) );
 			add_filter( $this->prefix . 'output_option', array( $this, 'display_custom_options' ), 10, 2 );
+			add_filter( $this->prefix . 'post_query', array( $this, 'filter_posts' ), 10, 1 );
+
 			add_action( $this->prefix . 'daily_update_cron', array( $this, 'daily_update' ) );
 			add_action( 'init', array( $this, 'make_dynamic_xsl' ) );
 			add_action( 'transition_post_status', array( $this, 'update_sitemap_from_posts' ), 10, 3 );
 			add_action( 'after_doing_aioseop_updates', array( $this, 'scan_sitemaps' ) );
 			add_action( 'all_admin_notices', array( $this, 'sitemap_notices' ) );
 		}
 
+		/**
+		 * Adds additional arguments before posts are fetched.
+		 *
+		 * @param array $args Arguments needed by get_posts/WP_Query.
+		 *
+		 * @return array
+		 */
+		function filter_posts( $args ) {
```

Comment:
This concept would probably be better where the filter is being applied since this seems like this would be core concept. Was there a reason for preferring an action hook?

https://github.com/semperfiwebdesign/all-in-one-seo-pack/blob/8bdb9d1c564ec607ae10dbe2821fa228e98d094b/modules/aioseop_sitemap.php#L4183

Reply:
the filter hook already existed and I'm merely using it. I don't see an action hook. Which one are you referring to?

Reply:
Your using a filter that's applied in a function that is only used by the sitemap module. Which makes the prefix for the filter redundant when it is just the sitemap module. This could be done the same way exclude is being done.

This would further spread out code, and it seems confusing; which the prefix itself makes it increasingly difficult to track.

Reply:
So I'm not a big fan of the way we use a variable prefix due to the decreased readability, but we do need to prefix them so there are no conflicts.

Reply:
@michaeltorbert The thing is, the function is only used by the sitemap module. There is no other module that uses that filter.

Reply:
@EkoJR Maybe we're not talking about the same thing. Are you talking about `add_filter( $this->prefix . 'post_query', array( $this, 'filter_posts' ), 10, 1 );` and recommending we use an unprefixed filter called post_query?

Reply:
@michaeltorbert No, but that is the filter. I was recommending not using the filter. Using it just seems increasingly complex; especially with the prefix.

If anything, a prepare function came to mind, but I'm not too sure about it.

Reply:
if there is a filter we should use it - its not only for other people to extend behavior but its for us to use it as well instead of using a prepared function. Any kind of hook does increase complexity (blame it on WP) but it also increases flexibility. I don't see any reason to not use the hook that already exists.